### PR TITLE
Enhanced login, sign up and forgot password pages

### PR DESCRIPTION
**Pull request by miyabitanimchi**
What I did on this branch is as follows:

- Added validation on login , sign up and forgot password pages (now error messages are shown on client too)
- Implemented a function to redirect to the browsing page (`/board`)
- Changed some style Kaho mentioned [here](https://github.com/shiv-chan/senpai-kohai/pull/5#pullrequestreview-811751367)
- some minor change, addition and deletion

Please check when you have time, thank you!

### CODE REVIEW

**Review by Dannyboy222 — APPROVED** [review]
👍

File: client/src/features/signup/SignUp.tsx [hunk 1]
```diff
@@ -39,50 +87,58 @@ const SignUp = () => {
...
               <input
+                id="email"
                 type="email"
-                className="mb-8 h-10 text-xl w-full pl-8"
-                onChange={(e) =>
-                  setInputs({ ...inputs, email: e.target.value })
-                }
+                className="h-10 text-xl w-full pl-8 rounded"
+                onChange={(e) => handleOnChange(e)}
+                onBlur={(e) => handleOnBlur(e)}
```
Comment (shiv-chan) [inline client/src/features/signup/SignUp.tsx:111]:
```suggestion
                onChange={handleOnChange}
                onBlur={handleOnBlur}
```
We can make it shorter like this. The same goes for line 130-131.

File: client/src/features/signup/SignUp.tsx [hunk 2]
```diff
@@ -1,36 +1,84 @@
...
+      } else {
+        console.error(error);
+        setErrMsgFromServer('Sorry, unexpected error occured');
+      }
+    }
+  };
+
+  const handleOnChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = e.target;
+    errMsgFromServer.length !== 0 && setErrMsgFromServer('');
```
Comment (shiv-chan) [inline client/src/features/signup/SignUp.tsx:66]: @miyabitanimchi I liked this 👀 💡

File: client/src/features/login/LogIn.tsx [hunk 3]
```diff
@@ -8,57 +8,71 @@ import { FaUnlockAlt } from 'react-icons/fa';
...
             <div className="relative">
               <input
+                id="email"
                 type="email"
-                className="mb-8 h-10 text-xl w-full pl-8"
-                onChange={(e) =>
-                  setInputs({ ...inputs, email: e.target.value })
-                }
+                className="mb-8 h-10 text-xl w-full pl-8 rounded "
+                onChange={(e) => handleOnChange(e)}
```
Comment (shiv-chan) [inline client/src/features/login/LogIn.tsx:73]:
```suggestion
                onChange={handleOnChange}
```
The same goes for line 87 and 132.

File: client/src/features/login/LogIn.tsx [hunk 4]
```diff
@@ -67,21 +81,20 @@ const LogIn = () => {
             </label>
             <div className="relative">
               <input
+                id="password"
                 type={isVisible ? 'text' : 'password'}
-                className="h-10 mb-12 w-full px-8"
-                onChange={(e) =>
-                  setInputs({ ...inputs, password: e.target.value })
-                }
+                className="h-10 mb-12 w-full px-8 rounded"
```
Comment (shiv-chan) [inline client/src/features/login/LogIn.tsx:86]: maybe it's better to have the same font size as email input.

File: client/src/features/forgotpassword/ForgotPasswordResetSuccess.tsx [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+const ForgotPasswordResetSuccess: React.FC = () => {
+  return (
+    <section className="flex justify-center my-auto h-9/12 w-1/2 tablet_l_max:w-full">
+      <div className="flex flex-col justify-center text-center h-full w-10/12 mobile_xl_max:w-9/12">
+        <p className="flex-initial text-2xl mb-10 mobile_l_max:text-xl">
+          New password set successfully!
+        </p>
+        <Link to="/login" className="flex-initial underline m-auto">
+          Login
```
Comment (shiv-chan) [inline client/src/features/forgotpassword/ForgotPasswordResetSuccess.tsx:14]: **Suggestion:** Back to login

**Review by shiv-chan — CHANGES_REQUESTED** [review]
Thank you for improving, @miyabitanimchi 🙏 

There's one thing I notice.
When you click "Send" button to reset a password with an empty input, the error message looks like the capture.
スクリーンショット 2021-12-09 22 19 01
Could you add the message for this case before merging?

Please check other comments too.

Thanks!

File: client/src/features/login/LogIn.tsx (hunk 4 above)
Comment (miyabitanimchi) [reply client/src/features/login/LogIn.tsx:86]: Thank you, I added text-xl for all the password inputs (`/login`, `/signup` and `forgotpassword/reset/:id`). 👍

**Comment by miyabitanimchi** [comment]
I improved code Kaho mentioned with the comments, and also added another error message when a user submit with empty value on `/forgotpassword` page. I added an error handling for that on `server/controllers/forgotPassword/forgotPassword.js`. 
Please review it again when you have time, and also let me know if I miss something and if you notice something 🙏

**Review by shiv-chan — APPROVED** [review]
It looks good to me ✨ 
Thanks for improving, @miyabitanimchi!

**Comment by shiv-chan** [comment]
@miyabitanimchi Go ahead and merge this to the target branch.
I updated some wordings. Please refer to [this commit](https://github.com/shiv-chan/senpai-kohai/pull/21/commits/11483cf6d1737ca1a7cc38a9234f13c1286b07c5).

Thanks!

**Pull request merged**